pipelines/rj_smas__disparo_pic/utils/discord.py

```
@@ -0,0 +1,225 @@
+        AND sendDate >= TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 3 HOUR)
+    )
+    SELECT
+      status,
+      COUNT(DISTINCT targetExternalId) as quantidade_cpfs,
+      COUNT(*) as quantidade_disparos,
+      ROUND(COUNT(DISTINCT targetExternalId) * 100.0 / SUM(COUNT(DISTINCT targetExternalId)) OVER(), 2) as percentual
+    FROM ranked_messages
+    WHERE rn = 1
+    GROUP BY status
+    ORDER BY quantidade_cpfs DESC
+    """
```

Ruff (0.14.1)

Possible SQL injection vector through string-based query construction

(S608)